### DISCUSSION TITLE
Supporting OpenAI Vision

### DISCUSSION THREAD

**Discussion by pj747** [langfuse/langfuse#982] [Ideas]
It would be awesome if langfuse could process images and calculate token cost for those as well.

**Comment by clemra** [comment]
Thank you for submitting this + looking forward to hearing from others about their requirements, too. 

Status Quo (as of today)
We support the vision API in a way that our (text-based) SDKs does not break:
- [base64 images](https://platform.openai.com/docs/guides/vision/uploading-base-64-encoded-images) will be filtered out of the OpenAI request, so that only your textual data will end up in langfuse
- [Image URLs](https://platform.openai.com/docs/guides/vision/quick-start) will be sent to langfuse and you can see the links in our UI

**Comment by techytushar** [comment]
It would be great if we can have an image preview feature when viewing a trace/session. We can have a small thumbnail image in the trace, which when clicked opens a full sized image in a new tab.

**Reply by marcklingen** [reply]
Langfuse now renders images within traces: https://langfuse.com/changelog/2024-08-21-openai-multi-modal-traces
